hashtypechecker.cpp

```diff
@@ -0,0 +1,99 @@
+#include "hashtypechecker.h"
+
+#include <QtDebug>
+
+HashTypeChecker::HashTypeChecker()
+{
+    connect(&m_john,SIGNAL(finished(int, QProcess::ExitStatus)),this,SLOT(startParsing()));
+    connect (&m_john, SIGNAL(readyReadStandardOutput()), this, SLOT(processOutput())); // connect process signals with your code
+}
+
+HashTypeChecker::~HashTypeChecker()
+{
+
+}
+void HashTypeChecker::start(QString& pathToJohn, QString& pathToPwdFile)
+{
+    m_john.start(pathToJohn + " --show=types " + pathToPwdFile);
+}
+void HashTypeChecker::terminate()
+{
+    if (m_john.state() != QProcess::NotRunning)
+        m_john.terminate();
+    m_john.waitForFinished(500);
+    if (m_john.state() != QProcess::NotRunning)
+        m_john.kill();
+
+    // Process what have been done so far from JohnOutput ...
+    startParsing();
+}
+void HashTypeChecker::startParsing()
+{
+    QtConcurrent::run(this,&HashTypeChecker::parseJohnAnswer);
```

Comment:
It's the way to say to QtConcurrent::run to use a member function. Without, this, it would need a static method. 
http://doc.qt.io/qt-4.8/qtconcurrentrun.html